cli/run.go

```diff
@@ -236,88 +292,205 @@ func run(c *cli.Context) error {
 	}
 
-	if err := registerSecretFiltering(enabledFiltering); err != nil {
-		failf("Failed to register Secret Filtering mode, error: %s", err)
+	enabledFiltering, err := isSecretFiltering(secretFiltering, inventoryEnvironments)
+	if err != nil {
+		return nil, fmt.Errorf("failed to check Secret Filtering mode: %s", err)
 	}
 
 	enabledEnvsFiltering, err := isSecretEnvsFiltering(secretEnvsFiltering, inventoryEnvironments)
 	if err != nil {
-		failf("Failed to check Secret Envs Filtering mode, error: %s", err)
+		return nil, fmt.Errorf("failed to check Secret Envs Filtering mode: %s", err)
 	}
 
-	if err := registerSecretEnvsFiltering(enabledEnvsFiltering); err != nil {
-		failf("Failed to register Secret Envs Filtering mode, error: %s", err)
+	noOutputTimeout := readNoOutputTimoutConfiguration(inventoryEnvironments)
+
+	return &RunConfig{
+		Modes: models.WorkflowRunModes{
+			CIMode:                  isCIMode,
+			PRMode:                  isPRMode,
+			DebugMode:               configs.IsDebugMode,
+			NoOutputTimeout:         noOutputTimeout,
+			SecretFilteringMode:     enabledFiltering,
+			SecretEnvsFilteringMode: enabledEnvsFiltering,
+		},
+		Config:   bitriseConfig,
+		Workflow: runParams.WorkflowToRunID,
+		Secrets:  inventoryEnvironments,
+	}, nil
+}
+
+func registerRunModes(modes models.WorkflowRunModes) error {
+	if err := registerCIMode(modes.CIMode); err != nil {
+		return fmt.Errorf("failed to register CI mode: %s", err)
 	}
 
-	isPRMode, err := isPRMode(prGlobalFlagPtr, inventoryEnvironments)
-	if err != nil {
-		failf("Failed to check PR mode, error: %s", err)
+	if err := registerPrMode(modes.PRMode); err != nil {
+		return fmt.Errorf("failed to register PR mode: %s", err)
 	}
 
-	if err := registerPrMode(isPRMode); err != nil {
-		failf("Failed to register PR mode, error: %s", err)
+	if err := registerSecretFiltering(modes.SecretFilteringMode); err != nil {
+		return fmt.Errorf("failed to register Secret Filtering mode: %s", err)
 	}
 
-	isCIMode, err := isCIMode(ciGlobalFlagPtr, inventoryEnvironments)
-	if err != nil {
-		failf("Failed to check CI mode, error: %s", err)
+	if err := registerSecretEnvsFiltering(modes.SecretEnvsFilteringMode); err != nil {
+		return fmt.Errorf("failed to register Secret Envs Filtering mode: %s", err)
 	}
 
-	if err := registerCIMode(isCIMode); err != nil {
-		failf("Failed to register CI mode, error: %s", err)
+	registerNoOutputTimeout(modes.NoOutputTimeout)
+	return nil
+}
+
+func createWorkflowRunPlan(targetWorkflow string, workflows map[string]models.WorkflowModel, uuidProvider func() string) models.WorkflowRunPlan {
+	var executionPlan []models.WorkflowExecutionPlan
+	workflowList := walkWorkflows(targetWorkflow, workflows, nil)
+	for _, workflowID := range workflowList {
+		workflow := workflows[workflowID]
+
+		var stepPlan []models.StepExecutionPlan
+		for _, stepItem := range workflow.Steps {
+			var stepID string
+			for id := range stepItem {
+				stepID = id
+			}
```

Comment:
I had a hard time understanding why this weird looking for loop is needed but now I know. This is because of the following:
```
steps:
- script:
    title: Run integration tests
    inputs:
    - content: |-
        #!/usr/bin/env bash
```
translates to a map where the key is `script` and value is a `StepModel` based on the other fields. And this map will always have a single item in it.

Then in go there is no simple way to do
```
stepID = map[map.keys[0]]
```
right?

Reply:
Yes, created a utility func (`StepListItemModel.GetStepIDAndStep`) to make it more readable.